testing/upload_validation.ts

```diff
@@ -1704,6 +1797,16 @@ function buildMetadataSchema({sdkVersion}: BuildMetadataSchemaArgs): {
         }
       });
     })
+    .superRefine((data, context) => {
+      const crawlHierarchy = getSyncTableHierarchy(data as PackVersionDefinition);
+      if (!crawlHierarchy) {
+        context.addIssue({
+          code: z.ZodIssueCode.custom,
+          path: ['syncTables'],
+          message: 'Sync table parent hierarchy is invalid',
```

Comment:
Can do later but ideally these messages are easily actionable, so would be good to pinpoint specific invalid cases, e.g. `crawlStrategy.parentTable "Issus" not found in this pack` and have the error path point at the table and parameter with the error.

So may want to reframe `getSyncTabeHierarchy` as `validateCrawlHierarchy` or something and pass the zod context so you can traverse using it.

Reply:
Great idea, done!